### ISSUE TITLE
Need help on query hive and bulkload to hbase

### ISSUE THREAD

**Issue by litao-buptsse** [unicredit/hbase-rdd#17]
I want query hive table and bulkload the result to hbase.
e.g. my hive table is like:

site | query | pv
--- | --- | ---
www.google.com | google | 100
www.google.com | google map | 99
github.com | github | 98
github.com | gist | 97

It's very easy for me to query the hive table using SparkSQL
```scala
val df = sqlContext.sql("select site, query, pv from tb")
```

My hbase is like follows, with site as row key, query as column and pv as value:
row key: $site
column family: cf
column: $query
value: $pv

I want to use toHBaseBulk() to load rdd data to hbase. I found that the rdd has to be RDD[(String, Map[String, A])], so I have to convert my rdd to this style.

I can use rdd transformation to convert like:
```scala
val rdd = df.map(r => (r.getString(0), (r.getString(1), Bytes.toBytes(r.getLong(2)))).
    groupByKey.map(r => (r._1, r._2.toMap))
```

So I can use toHBaseBulk() like:
```scala
rdd.toHBaseBulk(table, cf)
```

But it seems not efficient, because my query num is pretty huge under one site and I think the toMap will cause OOM. 

It's there any good solution for my problem?

**Comment by fralken** [comment]
you can try with

    val tohbase = rdd.map(r => (r.getString(0), (r.getString(1), Bytes.toBytes(r.getLong(2))))
      .groupByKey
      .mapValues(_.toMap)

also, instead of writing 

    rdd.map(r => (r.getString(0), (r.getString(1), Bytes.toBytes(r.getLong(2))))

you can define

    implicit val longWriter = new Writes[Long] {
      def write(data: Long) = Bytes.toBytes(data)
    }

and then

     rdd.map(r => (r.getString(0), (r.getString(1), r.getLong(2)))

**Comment by litao-buptsse** [comment]
@fralken Thank you very much, I will have a try.

**Comment by fralken** [comment]
It comes to my mind that maybe for your use case you don't even need to do a bulk load. You can use `rdd.toHBase()` instead. In this case you don't need to `groupByKey`.

    rdd.map(r => (r.getString(0), Map(r.getString(1) -> r.getLong(2)))
      .toHBase(....)

**Comment by litao-buptsse** [comment]
@fralken Thanks

**Comment by fralken** [comment]
@litao-buptsse We released 0.6.0 with a new implementation of bulk write. Now you can use `toHBase()` and `toHBaseBulk` interchangeably and see which better fits your needs. We also added implicit Reads and Writes for basic types, so you don't need to define `implicit val longWriter`since it's already available.

**Comment by litao-buptsse** [comment]
@fralken Thank you very much!
